src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -99,6 +101,10 @@ public static Transaction newChallenge(
         .addOperation(domainNameOperation)
         .addOperation(webAuthDomainOperation);
 
+    if (memo != null) {
+      builder.addMemo(memo);
```

Comment:
We need to ensure only memos of type ID are used here. Is there a subclass of memo for this type? Otherwise we should use a 64-bit integer as the parameter and create the memo object inside the function.

Reply:
Makes sense. I had added the type check 
```java

      if (!(memo instanceof MemoId)) {
        throw new InvalidSep10ChallengeException("only memo type `Id` is supported");
      }

```